packages/block-library/src/post-author/edit.js

```diff
@@ -1,25 +1,263 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
-import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import apiFetch from '@wordpress/api-fetch';
+import { useRef } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockColorsStyleSelector from './block-colors-selector';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+function PostAuthorDisplay( { props, author, authors } ) {
+	const ref = useRef();
+
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const { fontSize, setFontSize } = props;
+	const {
+		TextColor,
+		BackgroundColor,
+		InspectorControlsColorPanel,
+		ColorPanel,
+	} = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'background-color' },
+		],
+		{
+			contrastCheckers: [
+				{
+					backgroundColor: true,
+					textColor: true,
+					fontSize: fontSize.size,
+				},
+			],
+			colorDetector: { targetRef: ref },
+			colorPanelProps: {
+				initialOpen: true,
+			},
+		},
+		[ fontSize.size ]
+	);
+
+	const { align, id, showAvatar, showBio, byline } = props.attributes;
+
+	const avatarSizes = [
+		{ value: 24, label: __( 'Small' ) },
+		{ value: 48, label: __( 'Medium' ) },
+		{ value: 96, label: __( 'Large' ) },
+	];
```

Comment:
These sizes come from the following core function: https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api.php#L1211

As you can see, they can be changed using a filter, in which case this logic would break. Should we make that `/wp/v2/users/` request special in some way, so that the response is guaranteed to return correct avatar sizes?

Reply:
1. I 'll see how I can get those avatar sizes
2. I should probably check for everything that might not exist, it's just hard to know what that might be :) with all the filters we have.

Reply:
How about populating the avatar sizes dropdown with the keys of the `author.avatur_urls` object?

![image](https://user-images.githubusercontent.com/96308/80020518-7d1c5580-84d9-11ea-8f2b-85e46666fa29.png)

The dropdown could simply display dimensions, i.e.

```
24×24
48×48
96×96
```

Reply:
Well, I'm not sure about that. I believe 24x24 to be a terrible label for uses.

Reply:
Agree, in principle.

My priorities would be:
1. Avoid problems if some of [these sizes](https://github.com/WordPress/gutenberg/pull/19894/files#diff-7e82f718ff736f624069097db0995cb1R73) actually don't exist.
2. Implement better UI.

My strategy would thus be:
1. Show ugly size labels (`24×24` etc) for a first iteration to avoid problems with non-existing sizes.
2. In a follow-up PR, use a visual preview of the sizes, as seen e.g. here: https://github.com/WordPress/gutenberg/pull/19894#issuecomment-586376139

What do you think?

Reply:
OK, sounds like a good strategy. We can go then with an ugly UI and add a visual preview right after in the sidebar. Maybe I can do that already ...

Reply:
I added the simple size x size select labels.